plugins from babelOptions are applied differently in fixtures vs tests

- `babel-plugin-tester` version: 9.0.1
- `node` version: 14.3.0
- `npm` (or `yarn`) version: 1.22.4

Relevant code or config

```javascript
import pluginTester from 'babel-plugin-tester/pure';
import * as path from 'path';
import plugin from 'babel-plugin-macros';

pluginTester({
  plugin,
  babelOptions: {
    plugins: [
      [
        'babel-plugin-module-resolver',
        {
          alias: {
            '@material-ui/utils': path.resolve(__dirname, '../src'),
          },
        },
      ],
    ],
  },
  filename: __filename,
  fixtures: path.resolve(__dirname, './__fixtures__'),
  tests: [
    {
      title: 'can throw on missing error codes',
      error: /unknown: Missing error code for message 'missing'. Did you forget to run `yarn extract-errors` first?/,
      code: `
        import MuiError from '@material-ui/utils/MuiError.macro';

        throw new MuiError('missing');`,
      pluginOptions: {
        muiError: {
          missingError: 'throw',
        },
      },
    },
  ],
});

```

What you did:
- use `babel-plugin-module-resolver` in `babelOptions.plugins`

What happened:

- Tests in `tests` pass since `babel-plugin-module-resolver` is applied before the `plugin`.
- Tests in `fixtures` fail because `babel-plugin-module-resolver` is not applied before `plugin`
  <details>
  <summary>error message</summary>

  ```
  /home/eps1lon/Development/projects/mui/fork/packages/material-ui-utils/test/__fixtures__/literal/code.js: Cannot find module '@material-ui/utils/MuiError.macro' from '/home/eps1lon/Development/projects/mui/fork/packages/material-ui-utils/test/__fixtures__/literal'
  ```
  </details>

Reproduction repository: 
1. clone (checkout this specific commit) https://github.com/eps1lon/material-ui/tree/1ceb45f2f8ec509885fd68285f2ad90f8f1a4e44
2. `yarn`
3. `yarn cross-env NODE_ENV=test mocha packages/material-ui-utils/test/MuiError.macro.test.js --watch`


Problem description: I want to write the test that mirror actual usage. For babel macros I would need to use a different import source than what is actually used.

Suggested solution: 
1. Always apply `babelOptions.plugins` first
2. Always apply `babelOptions.plugins` last but add an option so that we can configure it (either flag "applyFirst" or custom merger function)

--- a/src/__tests__/plugin-tester.js
+++ b/src/__tests__/plugin-tester.js
@@ -43,6 +43,16 @@ const runPluginTester = async (...args) => {
   }
 }
 
+const pluginWithOrderTracking = (orderArray, orderInt) => () => {
+  return {
+    visitor: {
+      Program() {
+        orderArray.push(orderInt)
+      },
+    },
+  }
+}
+
 beforeEach(() => {
   equalSpy = jest.spyOn(assert, 'equal')
   errorSpy = jest.spyOn(console, 'error').mockImplementation(noop)
@@ -751,6 +761,40 @@ test('appends to root plugins array', async () => {
   expect(optionFoo).toHaveBeenCalledTimes(2)
   expect(optionBar).toHaveBeenCalledTimes(1)
   expect(programVisitor).toHaveBeenCalledTimes(15)
+})
+
+test('fixtures run plugins in the same order as tests', async () => {
+  const runOrder1 = []
+  const runOrder2 = []
+
+  await runPluginTester(
+    getOptions({
+      plugin: pluginWithOrderTracking(runOrder1, 2),
+      babelOptions: {
+        plugins: [
+          pluginWithOrderTracking(runOrder1, 1),
+          pluginWithOrderTracking(runOrder1, 3),
+        ],
+      },
+    }),
+  )
+
+  await runPluginTester(
+    getOptions({
+      plugin: pluginWithOrderTracking(runOrder2, 2),
+      tests: null,
+      fixtures: getFixturePath('creates-output-file'),
+      babelOptions: {
+        plugins: [
+          pluginWithOrderTracking(runOrder2, 1),
+          pluginWithOrderTracking(runOrder2, 3),
+        ],
+      },
+    }),
+  )
+
+  expect(runOrder1).toStrictEqual([1, 3, 2])
+  expect(runOrder2).toStrictEqual(runOrder1)
 })
 
 test('endOfLine - default', async () => {

--- a/src/plugin-tester.js
+++ b/src/plugin-tester.js
@@ -307,24 +307,28 @@ const createFixtureTests = (fixturesDir, options) => {
         ...rest
       } = options
 
-      const hasBabelrc = [
-        '.babelrc',
-        '.babelrc.js',
-        '.babelrc.cjs',
-      ].some(babelrc => fs.existsSync(path.join(fixtureDir, babelrc)))
+      const hasBabelrc = ['.babelrc', '.babelrc.js', '.babelrc.cjs'].some(
+        babelrc => fs.existsSync(path.join(fixtureDir, babelrc)),
+      )
 
       const {babelOptions} = mergeWith(
         {},
         fullDefaultConfig,
         {
           babelOptions: {
-            plugins: [[plugin, mergedFixtureAndPluginOptions]],
             // if they have a babelrc, then we'll let them use that
             // otherwise, we'll just use our simple config
             babelrc: hasBabelrc,
           },
         },
         rest,
+        {
+          babelOptions: {
+            // Ensure `rest` comes before `babelOptions.plugins` to preserve
+            // default plugin run order
+            plugins: [[plugin, mergedFixtureAndPluginOptions]],
+          },
+        },
         mergeCustomizer,
       )
 

